### PR TITLE
Oheger bosch/upgrade logging to log4j2

### PR DESCRIPTION
**Pull request by oheger-bosch**
This PR updates the logging framework from log4j 1.x to log4j2. The logging output should not be affected by this change.

The dependencies to log4j1 have been replaced by corresponding dependencies to log4j2; for some transitive dependencies to log4j1 explicit excludes have been added. While the log methods have not changed in the new version, the way how to obtain a logger is now different; also the logger classes have been moved to another package. Therefore, all the Java classes that make use of a logger had to be (slightly) adapted.

An initial XML-based configuration file for log4j2 has been added. It should be mainly equivalent to the existing configuration. The configuration could be enhanced later, e.g. to make use of advanced features like asynchronous logging. The properties files to configure log4j1 have been removed.

Note that this PR does not upgrade to the most recent version of log4j2. There is, unfortunately, an incompatibility of the Spring boot version in use with log4j2 versions greater than 2.7. More information can be found [here](https://stackoverflow.com/questions/50680649/log4j2-unrecognized-conversion-specifier-xwex-starting-at-position-160-in-co) and in [this ticket](https://github.com/spring-projects/spring-boot/issues/9172).

@blaumeiser-at-bosch @neubs-bsi 

### How To Test?
Deploy the changes in a running server and verify the log output.

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 import com.google.common.base.Strings;
 import com.google.common.collect.*;
 
+import org.apache.logging.log4j.LogManager;
```
Comment (blaumeiser-at-bosch) [inline backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java]: Would it make sense to have the imports sorted and close by each other? Please consider this in all cases.

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentSearchHandler.java [hunk 2]
```diff
@@ -9,11 +9,12 @@
  */
 package org.eclipse.sw360.datahandler.db;
 
+import org.apache.logging.log4j.LogManager;
```
Comment (blaumeiser-at-bosch) [inline backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentSearchHandler.java]: Import should be close to the other log import

File: backend/utils/src/main/java/org/eclipse/sw360/attachments/db/RemoteAttachmentDownloader.java [hunk 3]
```diff
@@ -50,36 +49,36 @@ public static int retrieveRemoteAttachments(Supplier<HttpClient> httpClient, Str
         AttachmentContentRepository attachmentContentRepository = new AttachmentContentRepository(new DatabaseConnector(httpClient, dbAttachments));
 
         List<AttachmentContent> remoteAttachments = attachmentContentRepository.getOnlyRemoteAttachments();
-        log.info(format("we have %d remote attachments to retrieve", remoteAttachments.size()));
+        log.info("we have {} remote attachments to retrieve", remoteAttachments.size());
```
Comment (neubs-bsi) [inline backend/utils/src/main/java/org/eclipse/sw360/attachments/db/RemoteAttachmentDownloader.java:52]: maybe remove the "we have". it is kind of weird

**Review by neubs-bsi — COMMENTED** [review]
I noticed that in some files you already changed the log messages to be formatted without a specific formatting function and nicely placing the objects at the end of the method. But in most files the log messages weren't changed. Is that on purpose?

File: backend/pom.xml [hunk 4]
```diff
@@ -48,9 +48,16 @@
...
-            <artifactId>log4j</artifactId>
-            <version>${log4j.version}</version>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
+        </dependency>
+        <dependency>
```
Comment (blaumeiser-at-bosch) [inline backend/pom.xml:58]: Where is the slf4j stuff needed?

**Review by blaumeiser-at-bosch — CHANGES_REQUESTED** [review]
Two small things

File: backend/pom.xml (hunk 4 above)
Comment (oheger-bosch) [reply backend/pom.xml:58]: Slf4j is still used by some parts of the codebase, also by the generated Thrift code and in transitive dependencies. This import makes sure that all log statements written through the slf4j API are routed to log4j.

File: backend/utils/src/main/java/org/eclipse/sw360/attachments/db/RemoteAttachmentDownloader.java (hunk 3 above)
Comment (oheger-bosch) [reply backend/utils/src/main/java/org/eclipse/sw360/attachments/db/RemoteAttachmentDownloader.java:52]: In the first 2 or 3 files I modified I adapted the log messages, but then noticed that placeholders are typically not used in the codebase. (Maybe this is one reason for the statement that logging slows down performance.) In this PR I would like to focus on the update of the logging dependency, so these changes are easier to review. The proper use of the logging API deserves a PR on its own.

**Review by blaumeiser-at-bosch — APPROVED** [review]

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java (hunk 1 above)
Comment (oheger-bosch) [reply backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java]: Logging-related imports have been ordered in all files that have been touched (unless I missed some ;-)

**Pull request merged**